### DISCUSSION TITLE
Dual-Stack: Access IPv6 addresses from pod

### DISCUSSION THREAD

**Discussion by slaecker** [k3s-io/k3s#9630] [Q&A] [answered]
Hi all - I'm having a hard time connecting to IPv6 addresses from pods. In some pod logs I noticed errors regarding connection timeouts to certain hostnames. I tracked this down to hosts resolving to IPv4 addresses work fine but IPv6 fails.

My Setup:
- K3s 1.28.6 on 1 server / 2 agents installed via K3s install script
- Server and agents have IPv4 (static) and IPv6 (dhcp) addresses assigned
- Server is installed with
```
--cluster-cidr=10.42.0.0/16,2001:cafe:42::/56 --service-cidr=10.43.0.0/16,2001:cafe:43::/112
--kubelet-arg node-ip=0.0.0.0
```
- Agents are installed with
```
--kubelet-arg node-ip=0.0.0.0
```
- Nodes have IPv4 and IPv6 addresses assigned
- IP forwarding is enabled on server and agents

```
net.ipv4.ip_forward = 1
net.ipv4.conf.all.forwarding = 1
net.ipv4.conf.default.forwarding = 1

net.ipv6.conf.all.forwarding = 1
net.ipv6.conf.default.forwarding = 1
```

When I'm starting an alpine test deployment and access the console I can ping IPv4 addresses normally but any IPv6 address fails.

Examples:

```
/ # ping amazon.de
PING amazon.de (176.32.108.185): 56 data bytes
64 bytes from 176.32.108.185: seq=0 ttl=239 time=35.165 ms
64 bytes from 176.32.108.185: seq=1 ttl=239 time=33.502 ms
64 bytes from 176.32.108.185: seq=2 ttl=239 time=33.723 ms
^C
--- amazon.de ping statistics ---
3 packets transmitted, 3 packets received, 0% packet loss
round-trip min/avg/max = 33.502/34.130/35.165 ms

/ # ping cloudflare.com
PING cloudflare.com (2606:4700::6810:85e5): 56 data bytes
```

Server logs don't show any issues regarding IPv6. Server and agents can perform both ping commands successfully, only from within the cluster it doesn't work.

Any hints are highly appreciated.

**Comment by brandond** [comment, marked as answer]
Have you seen: https://docs.k3s.io/installation/network-options?_highlight=flannel-ipv6-masq#flannel-options

**Reply by slaecker** [reply]
No, indeed I haven't seen this. Thanks a lot, it solved the issue.
